lib/dx/result.ex

```diff
@@ -145,12 +145,25 @@ defmodule Dx.Result do
   @doc """
   When given `{:ok, value, binds}` or `{:ok, value}`, returns `value`.
   Otherwise, raises an exception.
+
+  ## Examples
+
+      iex> Dx.Result.unwrap!({:error, %ArgumentError{}})
+      ** (ArgumentError) argument error
+
+      iex> Dx.Result.unwrap!({:error, :timeout})
+      ** (Dx.Error.Timeout) A timeout occurred while loading the data required.
+
+      iex> Dx.Result.unwrap!({:error, :not_an_exception})
+      ** (Dx.Error.Generic) Error occurred: :not_an_exception
   """
   def unwrap!({:ok, result}), do: result
   def unwrap!({:ok, result, _binds}), do: result
   def unwrap!({:not_loaded, _data_reqs}), do: raise(Dx.Error.NotLoaded)
   def unwrap!({:error, {e, stacktrace}}), do: reraise(e, stacktrace)
```

Comment:
While you're adding guards, could you please add two to this clause as well? `is_exception` and `is_list` should be the correct ones. I think.

Reply:
I checked, `is_list` should be good for the stacktrace.